### PR TITLE
Feature/109823 befe introduce property on formlink type

### PR DESCRIPTION
**Pull request by Tom-Ritense**
Added new toggle to Form selection screen.

### CODE REVIEW

File: projects/valtimo/process-link/src/lib/components/process-link-modal/process-link-modal.component.html [hunk 1]
```diff
@@ -74,6 +74,13 @@ <h3 cdsModalHeaderHeading>
         *ngIf="obs.currentStepId === 'selectForm' || obs.typeOfSelectedProgressLink === 'form'"
       >
         <valtimo-select-form></valtimo-select-form>
+        <br/>
+        <cds-toggle
+          [label]="'processLinkConfiguration.toggleViewModel.title' | translate"
+          [onText]="'processLinkConfiguration.toggleViewModel.checkboxDescription' | translate"
+          [offText]="'processLinkConfiguration.toggleViewModel.checkboxDescription' | translate"
+          [checked]="viewModelEnabled?.value"
```
Comment (Tom-Ritense) [inline projects/valtimo/process-link/src/lib/components/process-link-modal/process-link-modal.component.html]: @teodor-ritense could you help us and tell how to bind this field to the selectedProcessLink. @sirolf2009 will be taking over this PR

File: projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts [hunk 2]
```diff
@@ -61,7 +57,8 @@ export class SelectFormComponent implements OnInit, OnDestroy {
     private readonly stateService: ProcessLinkStateService,
     private readonly processLinkService: ProcessLinkService,
     private readonly buttonService: ProcessLinkButtonService
-  ) {}
+  ) {
+  }
```
Comment (teodor-ritense) [inline projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts]:
```suggestion
  ) {}
```

File: projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts [hunk 3]
```diff
@@ -110,10 +107,13 @@ export class SelectFormComponent implements OnInit, OnDestroy {
...
 
   private updateProcessLink(): void {
-    this.stateService.selectedProcessLink$.pipe(take(1)).subscribe(selectedProcessLink => {
+    combineLatest([this.stateService.selectedProcessLink$, this.stateService.viewModelEnabled$])
+      .pipe(take(1))
+      .subscribe(([selectedProcessLink, viewModelEnabled]) => {
       const updateProcessLinkRequest: FormProcessLinkUpdateRequestDto = {
         id: selectedProcessLink.id,
         formDefinitionId: this._selectedFormDefinition.id,
+        viewModelEnabled: viewModelEnabled
```
Comment (teodor-ritense) [inline projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts]: This has the same effect

```suggestion
        viewModelEnabled
```

File: projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts [hunk 4]
```diff
@@ -128,17 +128,19 @@ export class SelectFormComponent implements OnInit, OnDestroy {
...
-            activityId: modalParams.element.id,
-          })
+        switchMap(([modalParams, processLinkTypeId, viewModelEnabled]) => {
+            return this.processLinkService.saveProcessLink({
+              formDefinitionId: this._selectedFormDefinition.id,
+              activityType: modalParams.element.activityListenerType,
+              processDefinitionId: modalParams.processDefinitionId,
+              processLinkType: processLinkTypeId,
+              activityId: modalParams.element.id,
+              viewModelEnabled: viewModelEnabled
```
Comment (teodor-ritense) [inline projects/valtimo/process-link/src/lib/components/select-form/select-form.component.ts]:
```suggestion
              viewModelEnabled
```

File: projects/valtimo/process-link/src/lib/services/process-link-state.service.ts [hunk 5]
```diff
@@ -116,6 +117,14 @@ export class ProcessLinkStateService implements OnDestroy {
     this.processLinkStepService.setProcessLinkTypeSteps(processLinkTypeId, hasOneOption);
   }
 
+  setViewModelEnabled(viewModelEnabled: boolean): void {
```
Comment (teodor-ritense) [inline projects/valtimo/process-link/src/lib/services/process-link-state.service.ts]:
```suggestion
  public setViewModelEnabled(viewModelEnabled: boolean): void {
```

File: projects/valtimo/process-link/src/lib/services/process-link-state.service.ts [hunk 6]
```diff
@@ -116,6 +117,14 @@ export class ProcessLinkStateService implements OnDestroy {
...
+    this._viewModelEnabled$.next(viewModelEnabled);
+  }
+
+  get viewModelEnabled$(): Observable<boolean> {
```
Comment (teodor-ritense) [inline projects/valtimo/process-link/src/lib/services/process-link-state.service.ts]:
```suggestion
  public get viewModelEnabled$(): Observable<boolean> {
```

**Pull request merged**